### PR TITLE
Post ride

### PR DESCRIPTION
**Pull request by oriefrati**
## ℹ️ Change description:
Nothing here

<details><summary><sub>templates<sub></summary>
<p>
  
<sub>[templates: [bug](?expand=1&template=bug.md), [feature](?expand=1&template=feature.md)]</sub>

</p>
</details>

### CODE REVIEW

File: examples/client/Locomotion/src/Components/SelectableButton/index.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from 'react';
+import { View } from 'react-native';
+import propsTypes from 'prop-types';
+import Loader from '../Loader';
+import { ButtonTextContainer, StyledButton, SubmitButtonText } from './styled';
+
+const RoundedButton = ({
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/SelectableButton/index.js]: change name

File: examples/client/Locomotion/src/Components/TextInput/index.js [hunk 2]
```diff
@@ -14,14 +17,41 @@ const TextInput = (props) => {
...
-// width: ${({ width }) => (width || '100%')};
-// border-bottom-width: 1px;
-// margin: 15px auto;
-// ${({ theme }) => `
-//     border-bottom-color: ${theme.isDarkMode ? '#fff' : '#e2e2e2'};
-//     color: ${theme.textColor};
-//     ${commonInputStyleWithTheme(theme)}
-//   `}
-// `;
+export const TextInputWithIcon = forwardRef((props, ref) => {
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/TextInput/index.js]: should be in different folder, not on the same file

File: examples/client/Locomotion/src/context/newRideContext/api.js [hunk 3]
```diff
@@ -35,6 +35,16 @@ export const createRide = async (ride) => {
   return data;
 };
 
+export const patchRide = async (rideId, newData) => {
+  try {
+    const { data } = await network.patch(`api/v1/rides/${rideId}`, newData);
+    return data;
+  } catch (e) {
+    console.error(e);
+    throw new Error(e);
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/context/newRideContext/api.js]: throw e

File: examples/client/Locomotion/src/context/newRideContext/index.tsx [hunk 4]
```diff
@@ -111,6 +114,8 @@ export const RidePageContext = createContext<RidePageContextInterface>({
   setServiceRequestFailed: () => undefined,
   ride: {},
   trackRide: async () => '',
+  postRideSubmit: (rideId: string, rating: number | null, tip: number | null) => undefined,
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/context/newRideContext/index.tsx:117]: rating?: number, tip?: number

File: examples/client/Locomotion/src/context/rides/api.js [hunk 5]
```diff
@@ -29,8 +29,8 @@ export const cancelRideApi = async () => {
   await network.post('api/v1/me/rides/cancel-active-ride');
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/context/rides/api.js]: I think we can delete this file

File: examples/client/Locomotion/src/pages/PostRide/StarRating.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    setRating(rate);
+  };
+
+  useEffect(() => {
+    onUpdate(rating);
+  }, [rating]);
+
+  return (
+    <SummaryStars>
+      <Star isOn={rating >= 1} onPress={() => updateRating(1)} />
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/PostRide/StarRating.js]:
```const starsArray = new array().fill(5);
{starsArray.map((star, index) => {
      <Star isOn={rating >= index + 1} onPress={() => updateRating(index + 1)} />
})}
}```

File: examples/client/Locomotion/src/pages/PostRide/Tips/CustomTip.js [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+      </DetailsContainer>
+      <DetailsContainer>
+
+        <ErrorContainer>
+          {!isValid && isValid !== null ? <ErrorText>Please enter a valid amount</ErrorText> : null}
+        </ErrorContainer>
+      </DetailsContainer>
+
+      <DetailsContainer style={{ marginTop: 15 }}>
+        <View style={{ flex: 1, paddingRight: 10 }}>
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/PostRide/Tips/CustomTip.js]: change to styled component

File: examples/client/Locomotion/src/pages/PostRide/Tips/CustomTip.js [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+          }}
+          onFocus={(e) => {
+            expand();
+          }}
+        />
+      </DetailsContainer>
+      <DetailsContainer>
+
+        <ErrorContainer>
+          {!isValid && isValid !== null ? <ErrorText>Please enter a valid amount</ErrorText> : null}
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/PostRide/Tips/CustomTip.js]: use i18n

File: examples/client/Locomotion/src/context/newRideContext/index.tsx (hunk 4 above)
Comment (oriefrati) [reply examples/client/Locomotion/src/context/newRideContext/index.tsx:117]: they are required

File: examples/client/Locomotion/src/pages/PostRide/Tips/CustomTip.js [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+const isValidNumber = async (number) => {
+  try {
+    const newNumber = await numberSchema.validate(number);
+    return newNumber;
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+const numberSchema = yup.number().required().positive()
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/PostRide/Tips/CustomTip.js]: move up

File: examples/client/Locomotion/src/pages/PostRide/Tips/index.js [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+  const serviceDisplayPrice = getSymbolFromCurrency(priceCurrency || 'USD');
+  const tipSuffix = isPercentage ? '%' : serviceDisplayPrice;
+
+  const bottomSheetRef = useRef(null);
+  const {
+    setSnapPointsState,
+  } = useContext(BottomSheetContext);
+
+  useEffect(() => {
+    setSnapPointsState(['70%', '100%']);
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/PostRide/Tips/index.js]: use const, we have a map inside the bottom sheet context

File: examples/client/Locomotion/src/pages/PostRide/Tips/index.js [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+
+  const calculateTipAmount = () => {
+    let calculatedTip = 0;
+    if (!customTip && !selectedTip) {
+      return calculatedTip;
+    }
+
+    calculatedTip = customTip || selectedTip;
+    if (isPercentage) {
+      calculatedTip = ridePrice * (customTip || selectedTip) / 100;
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/PostRide/Tips/index.js]: calculatedTip = ridePrice * (calculatedTip) / 100;

File: examples/client/Locomotion/src/pages/PostRide/index.js [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+        onIconPress={() => navigation.navigate(MAIN_ROUTES.HOME)}
+        iconSide={menuSide}
+      />
+      <PageContent>
+        <RatingContainer>
+          <SummaryStarsTitle>{i18n.t('postRide.ratingHeadline')}</SummaryStarsTitle>
+          <StarRating onUpdate={onRatingUpdate} />
+        </RatingContainer>
+
+        <TipsContainer style={{
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/PostRide/index.js]: add the style into the styled component object

File: examples/client/Locomotion/src/pages/PostRide/styled.js [hunk 13]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import {
+  View, Text, Image, TouchableOpacity,
+} from 'react-native';
+import styled from 'styled-components';
+import xIcon from '../../assets/x.png';
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/PostRide/styled.js]: use svg, you can cancel.svg inside bottomSheet, you can change the name to x and move it out

**Pull request merged**